doc/main.qbk

```diff
@@ -52,44 +59,122 @@
 read performance on modern SSDs or equivalent high-IOPS devices. The most
 common application for NuDB is content addressible storage where a
 cryptographic digest of the data is used as the key. The read performance
-and memory usage are independent of the size of the database. These are
-some other features:
+and memory usage are independent of the size of the database.
 
-[heading History]
+[heading Example]
 
-The first versions of rippled, the application behind the Ripple consensus
-network, used SQLite as their back end for unstructured data. The
-performance quickly became a limiting factor.
+This complete program creates a database, opens the database, inserts several
+key/value pairs, fetches the key/value pairs, closes the database, then erases
+the database files. Source code for this program is located in the examples
+directory.
 
-Then rippled then went through a series of back ends including LMDB, LevelDB, and
-RocksDB. Each of these databases performed well at first, but as the data
-size increased, memory usage increased and performance dropped off drastically.
+[import ../examples/example.cpp]
+[main]
+
+[variablelist Notes
+[
+    [(1) Files]
+    [
+    A database is represented by three files: the data file, the key file,
+    and the log file. Each file has a distinct header in a well known format.
+    The data file holds all of the key/value pairs and is serially iterable. The
+    key file holds a hash table indexing all of the contents in the data file.
+    The log file holds information used to roll the database back in the event
+    of a failure.
+    ]
+]
+[
+    [(2) Create a new database]
+    [
+    The `create` function creates a new data file and key
+    file for a database with the specified parameters. The caller specifies
+    the hash function to use as a template argument, the file paths,
+    and the database constants.
+    ]
+]
+[
+    [(3) Open an existing database]
+    [
+    The `open` function prepares a database for
+    insertion and/or reading.
+    ]
+]
+[
+    [(4) Insert a key/value pair]
+    [
+    Once a database is open, it becomes possible to insert new key/value pairs.
+    If the key already exists, the error is set to
+    [link nudb.ref.nudb__error.key_exists error::key_exists]. All keys in a NuDB
+    database must be unique. Multiple threads can call insert at the same time.
+    Internally however, insertions are serialized to present a consistent view
+    of the database to callers.
+    ]
+]
+[
+    [(5) Fetch a value given it's key]
+    [
+    The fetch function takes a callback object as a parameter which is
+    invoked when the value is retrieved.
+    ]
+]
+[
+    [(6) Terminate access to the database]
+    [
+    All resources used by the database
+    structures are freed.
+    ]
+]
+[
+    [(7) Delete the database]
+    [
+    To delete the database, delete the triplet of files which define it.
+    ]
+]
+]
```

Comment:
But that would make the source code a lot longer.  I think it's just right as it is.  The example is used to drive the narrative.  Also think about what people are going to do.  They'll copy the example into their own project, move some stuff around, and compile.  We don't want to discourage this.